pallets/did/src/lib.rs

```diff
@@ -28,93 +28,345 @@ mod tests;
 pub mod benchmarking;
 
 pub mod default_weights;
-pub mod migration;
-pub use default_weights::WeightInfo;
+// pub mod migration;				// Temporary disabled
 
 use codec::{Decode, Encode};
-use frame_support::{
-	decl_event, decl_module, decl_storage, dispatch::DispatchResult, Parameter, StorageMap,
-};
+pub use default_weights::WeightInfo;
+
+use frame_support::{StorageMap, decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult, ensure};
 use frame_system::{self, ensure_signed};
-use sp_runtime::{codec::Codec, traits::Member};
-use sp_std::prelude::*;
+use sp_runtime::traits::Verify;
+use sp_std::{prelude::{Clone}, convert::TryFrom, collections::{btree_set::BTreeSet}, vec::Vec};
+use sp_core::{ed25519, sr25519};
+
+/// Type of a payload of data (to verify signatures against).
+pub type Payload = Vec<u8>;
```

Comment:
Clippy suggests to replace Vec<u8> with &[8] (with a 'static lifetime parameter). Nevertheless, if I do that, Substrate complains with the following error: `the trait _::_parity_scale_codec::WrapperTypeDecode is not implemented for &[u8]`, so I am not sure whether we should ignore this Clippy warning or work to fix it?